CONTRIBUTING.md

```diff
@@ -1,35 +1,153 @@
+cargo fmt --all
+# Run Clippy (note that some lints are ignored, so `cargo clippy` will give many false positives)
+make clippy
+```
+
+See the [style doc](https://github.com/rust-lang/rfcs/blob/master/style-guide/README.md) and the [API guidelines](https://rust-lang-nursery.github.io/api-guidelines/) for details on the conventions.
+
+Please follow this style to make TiKV easy to review, maintain, and develop.
+
+
+### Build issues
+
+To reduce compilation time, TiKV builds do not include full debugging information by default &mdash; `release` and `bench` builds include no debuginfo; `dev` and `test` builds include line numbers only. The easiest way to enable debuginfo is to precede build commands with `RUSTFLAGS=-Cdebuginfo=1` (for line numbers), or `RUSTFLAGS=-Cdebuginfo=2` (for full debuginfo). For example,
+
+```bash
+RUSTFLAGS=-Cdebuginfo=2 make dev
+RUSTFLAGS=-Cdebuginfo=2 cargo build
+```
+
+When building with make, cargo will automatically use [pipelined][p] compilation to increase the parallelism of the build. To turn on pipelining while using cargo directly, set `CARGO_BUILD_PIPELINING=true`.
+
+[p]: https://internals.rust-lang.org/t/evaluating-pipelined-rustc-compilation/10199
+
+
+## Running TiKV
+
+To run TiKV as an actual key-value store, you will need to run it as a cluster (a cluster can have just one node, which is useful for testing). You can do this on a single machine or on multiple machines. You need to use [PD](https://github.com/pingcap/pd) to manage the cluster (even if there is just one node on a single machine). Instructions are in our [docs](docs/how-to/deploy/using-binary.md) (if you build TiKV from source, then you don't need to download the binary).
+
+
+### Configuration
+
+Read our configuration guide to learn about various [configuration options](./docs/reference/configuration). There is also a [configuration template](./etc/config-template.toml).
 
-- Fork the repository on GitHub.
-- Read the README.md for build instructions.
-- Play with the project, submit bugs, submit patches!
 
 ## Contribution flow
 
 This is a rough outline of what a contributor's workflow looks like:
 
-- Create a topic branch from where you want to base your work. This is usually master.
-- Make commits of logical units and add test case if the change fixes a bug or adds new functionality.
-- Run tests and make sure all the tests are passed.
-- Make sure your commit messages are in the proper format (see below).
-- Push your changes to a topic branch in your fork of the repository.
-- Submit a pull request.
-- Your PR must receive LGTMs from two maintainers.
+- Create a Git branch from where you want to base your work. This is usually master.
+- Write code, add test cases, and commit your work (see below for message format).
+- Run tests and make sure all tests pass.
+- Push your changes to a branch in your fork of the repository and submit a pull request.
+- Your PR will be reviewed by two maintainers (who may request some changes).
```

Comment:
```suggestion
- Your PR is required to be reviewed by at least two maintainers, who may request some changes.
```

Reply:
These changes make review sound like a barrier, rather than a helpful discussion, which is something I'm trying to avoid with this PR

Reply:
Indeed, although it's stating the fact. If the review is initiated by maintainers, your original wording is fine (I first thought the contributor needs to assign the reviewers - but they might not know any). But we can remove the parenthesis.